pilot/pkg/config/monitor/copilot_snapshot.go

```diff
@@ -96,7 +90,14 @@ func (c *CopilotSnapshot) ReadConfigFiles() ([]*model.Config, error) {
 			vs = createVirtualService(gatewayNames, route)
 		}
 
-		vs.Http = append(vs.Http, cr)
+		cr := createRoute(route)
+		cr.Route[0].Destination.Subset = route.GetCapiProcessGuid()
+		if route.GetPath() != "" {
+			cr.Match = createMatchRequest(route)
+			vs.Http = append([]*networking.HTTPRoute{cr}, vs.Http...)
+		} else {
+			vs.Http = append(vs.Http, cr)
```

Comment:
aren't this line and line 97 duplicates? I think we can rewrite this whole block to:

```go
cr := createRoute(route)
cr.Route[0].Destination.Subset = route.GetCapiProcessGuid()
if route.GetPath() != "" {
    cr.Match = createMatchRequest(route)
}
vs.Http = append(vs.Http, cr)
```

Or was the goal to differentiate `append(cr, vs.Http...)` vs `append(vs.Http, cr)`? If the goal is to fix that discrepancy, then I think you'd be better off sorting the HTTP routes by these conditions before returning it, rather than keeping the ordering as we process the list (IMO it's more obvious, and it'll be easier to change later if match ordering needs to change for some other reason down the line).

Reply:
We considered this but then you have the mess of what is potentially a nil pointer (there may not be a match for the route you are looking at).

Reply:
IMO sorting is a cleaner/better way to implement this because it consolidates the logic into a single location and it's fairly obvious what's happening. The impl as-is is a big tripping hazard IMO: that logic diff is subtle and easy to change later. Either way, doesn't warrant blocking the PR, though I would recommend adding a comment describing what's happening here.